jsonschema/validators.py

```diff
@@ -568,17 +652,6 @@ def base_uri(self):
         uri, _ = urldefrag(self.resolution_scope)
         return uri
 
-    @contextlib.contextmanager
```

Comment:
Is there a reason this is being removed?

Reply:
I think I removed this in order to increase the coverage as this code is not used anywhere in the library. I can create a dedicated PR for this or simply revert it if this is a common use case of the lib.

Reply:
It's part of the public API of the object, so we can't remove it without deprecating it.

Reply:
I agree, this can be done in another PR. I restored the method.